### PR TITLE
feat: 1044 Add cancel sell order action

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1044

image

Add cancel sell order action to the sell orders table.
This will be the last PR targeting `feat-985-sell-order-table`.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/pages/Marketplace/Storefront/Storefront.mock.ts [hunk 1]
```diff
@@ -1,75 +1,83 @@
-import { SellOrder } from './Storefront.types';
+import { longify } from '@cosmjs/stargate/build/queries/utils';
```
Comment (mhagel) [inline web-registry/src/pages/Marketplace/Storefront/Storefront.mock.ts:1]: Nice, had no idea this existed!

**Review by mhagel — APPROVED** [review]

File: web-registry/src/pages/Marketplace/Storefront/Storefront.mock.ts [hunk 2]
```diff
@@ -1,75 +1,83 @@
...
+import { SellOrderInfo } from '@regen-network/api/lib/generated/regen/ecocredit/marketplace/v1/query';
 
-export const sellOrdersMock: SellOrder[] = [
+export const sellOrdersMock: SellOrderInfo[] = [
```
[the remark this answers is not in the document]
Comment (victordelval) [reply web-registry/src/pages/Marketplace/Storefront/Storefront.mock.ts:4]: Very nice!

**Review by victordelval — APPROVED** [review]
Looks very good, thanks!

**Comment by erikalogie** [comment]
This looks great, a couple of small pieces of feedback:
- https://monosnap.com/file/HjCfk6zIBFMvaPm15ASPr7wQYAygoP
- https://monosnap.com/file/8s5MvcW7hpHQfgb1jaspwcefg8Axjv

**Pull request merged**